src/controllers/Student.controller.js

```diff
@@ -0,0 +1,40 @@
+const bcrypt = require('bcryptjs');
+const Student = require('../models/Student.model.js');
+
+const signup = (req, res) => {
+  const {
+    username,
+    password,
+    age,
+    gender,
+    english,
+    postcode,
+  } = req.body;
+
+  bcrypt.hash(password, 10, (hashError, hashedPassword) => {
```

Comment:
Use promises in the student function.
I mean, try to create separate functions to make the code clear.